client/src/pages/queries/components/chart-detail.tsx

```diff
@@ -0,0 +1,284 @@
+      },
+    },
+  )
+
+  const { mutate: removeChart, isLoading: isRemovingChart } = useMutation(
+    deleteChart,
+    {
+      onMutate: (deletedChart) => {
+        queryClient.setQueryData(
+          ['charts', queryId],
+          charts?.filter((chart) => chart.id !== deletedChart.id),
+        )
+        queryClient.removeQueries(['chart', chartId])
+        const updatedCharts: Chart[] | undefined = queryClient.getQueryData([
+          'charts',
+          queryId,
+        ])
+        changeActiveChartId(updatedCharts?.length ? updatedCharts[0].id : -1)
+      },
+    },
+  )
+
+  const onFinish = useCallback(
+    (values: any) => {
+      editChart({
+        id: chartId,
+        ...values,
+      })
+    },
+    [editChart, chartId],
+  )
+
+  const confirmDelete = useCallback(() => {
+    Modal.confirm({
+      title: 'Delete this chart?',
+      content: 'This action cannot be reverted',
+      okText: 'Yes',
+      okType: 'danger',
+      cancelText: 'No',
+      onOk() {
+        removeChart({ id: chartId })
+      },
+    })
+  }, [removeChart, chartId])
+
+  const chartData = useMemo(() => {
+    if (label && values && queryResult) {
+      return values
+        .map((value) =>
+          queryResult.rows.map((row) => {
+            return { label: row[label], value: row[value], type: value }
+          }),
+        )
+        .flat()
+    }
+  }, [queryResult, label, values])
+
+  const chartContent = useMemo(() => {
+    if (isLoading || !chartData) {
+      return (
+        <div className="flex flex-1 w-full h-full">
+          <div className="flex flex-col w-full px-4 pb-8 space-y-4">
+            <div className="w-48 h-8 bg-background-secondary animate-pulse" />
+            <div className="w-full h-full bg-background-secondary animate-pulse" />
+          </div>
+          <div className="flex-shrink-0 w-64 h-full pt-10 pl-4 space-y-8 border-l">
+            {range(5).map((val) => (
+              <div
+                key={val}
+                className="w-full h-8 bg-background-secondary animate-pulse"
+                style={{ opacity: 1 - val / 5 }}
+              />
+            ))}
+          </div>
+        </div>
+      )
+    }
+
+    if (error) {
+      return <Result status="warning" subTitle={(error as any).message} />
+    }
+
+    if (chart) {
+      return (
+        <div
+          className="flex w-full h-full"
+          style={{ width: 'calc(100% - 16rem)' }}
+        >
+          <div
+            className="flex flex-col px-4 pb-8 space-y-4"
+            style={{ width: 'calc(100% - 16rem)' }}
+          >
+            <div className="font-semibold">{title}</div>
+            {cloneElement(chartList[chartType].chart, { data: chartData })}
+          </div>
+          <Form
+            layout="vertical"
+            className="flex-shrink-0 w-64 h-full pl-4 border-l"
+            form={form}
+            onValuesChange={() => setDisbaled(false)}
```

Comment:
Instead of setting disabled, we can use `form.isFieldsTouched()`. It would return true if the form values are updated.

Reply:
So instead of having states and form for state management, we can use this example https://ant.design/components/form/#components-form-demo-customized-form-controls, to only use state management.